src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseImportService.java

```diff
@@ -121,6 +144,204 @@ public void importRepositories(final ProgrammingExercise templateExercise, final
+                copyExerciseContentToRepositoryForJavaOrKotlinExercise(testRepo, RepositoryType.TESTS, basePath, newExercise, oldPackageName);
+            }
+            case C, VHDL, ASSEMBLER -> {
+                copyExerciseContentToRepositoryForExerciseWithPlainDirectoryStructure(templateRepo, RepositoryType.TEMPLATE, basePath);
+                copyExerciseContentToRepositoryForExerciseWithPlainDirectoryStructure(solutionRepo, RepositoryType.SOLUTION, basePath);
+                copyExerciseContentToRepositoryForExerciseWithPlainDirectoryStructure(testRepo, RepositoryType.TESTS, basePath);
+            }
+            case PYTHON -> {
+                copyExerciseContentToRepositoryForPythonExercise(templateRepo, RepositoryType.TEMPLATE, basePath);
+                copyExerciseContentToRepositoryForPythonExercise(solutionRepo, RepositoryType.SOLUTION, basePath);
+                copyExerciseContentToRepositoryForPythonExercise(testRepo, RepositoryType.TESTS, basePath);
+            }
+            case OCAML -> {
+                copyExerciseContentToRepositoryForOcamlExercise(templateRepo, RepositoryType.TEMPLATE, basePath);
+                copyExerciseContentToRepositoryForOcamlExercise(solutionRepo, RepositoryType.SOLUTION, basePath);
+                copyExerciseContentToRepositoryForOcamlExercise(testRepo, RepositoryType.TESTS, basePath);
+            }
+            case HASKELL -> {
+                copyExerciseContentToRepositoryForHaskellExercise(templateRepo, RepositoryType.TEMPLATE, basePath);
+                copyExerciseContentToRepositoryForHaskellExercise(solutionRepo, RepositoryType.SOLUTION, basePath);
+                copyExerciseContentToRepositoryForHaskellExercise(testRepo, RepositoryType.TESTS, basePath);
+            }
+        }
+    }
+
+    private void copyExerciseContentToRepositoryForExerciseWithPlainDirectoryStructure(Repository repository, RepositoryType repositoryType, Path exerciseDir) throws IOException {
+        for (Path file : retrieveRepositoryContentPaths(retrieveRepositoryDirectoryPath(exerciseDir, repositoryType.getName()))) {
+            repositoryService.createFile(repository, String.valueOf(file.getFileName()), Files.newInputStream(file));
+
+        }
+
+    }
+
+    private void copyExerciseContentToRepositoryForPythonExercise(Repository repository, RepositoryType repositoryType, Path exerciseDir) throws IOException {
+        boolean testRepo;
+        if (repositoryType == RepositoryType.TESTS) {
+            repositoryService.createFolderRecursively(repository, "behavior");
+            repositoryService.createFolderRecursively(repository, "structural");
+            testRepo = true;
+        }
+        else {
+            testRepo = false;
+        }
+
+        for (Path file : retrieveRepositoryContentPathsPython(exerciseDir, repositoryType)) {
+            if (testRepo) {
+                if (file.toAbsolutePath().toString().contains("behavior")) {
```

Comment:
What happens if someone defines a structural test case file `test_behavior.py`?

Idea: You could instead use the `file.iterator()` in a helper method to check if any path segment matches `behavior` and if it is a directory.